Lib/sysconfig.py

```diff
@@ -3,6 +3,7 @@
 import os
 import sys
 from os.path import pardir, realpath
+from distutils.errors import DistutilsPlatformError
```

Comment:
This is a problem.

Reply:
We agreed with @encukou that it has to be removed eventually but it doesn't have to be done in this PR. The exception is used only in `get_python_inc` and `get_python_lib` which are both new functions moved from `distutils.sysconfig` so a possible solution might be to let them return `None` in case the functions are not able to return correct paths and wrap them in original `distutils.sysconfig` module to raise this exception if they return `None`. What do you think? They can also raise a different exception instead of returning None.

Reply:
Could you keep the two functions in distutils?

Reply:
Will these two functions be removed together with the module? It's the case for customize_compiler, which is also deprecated together with the module. `get_python_lib` seems to be useless for stdlib and `get_python_inc` seems to be also used only in distutils for building modules so it should be okay to move them back and deprecate them. Is that right?

Reply:
> What do you think? They can also raise a different exception instead of returning None.

I quite like this idea. I'm also OK with actually defining an exception class here, and inheriting from it in distutils.

Reply:
> Will these two functions be removed together with the module?

I meant in distutils, in one of the submodules that are not sysconfig :slightly_smiling_face:

Reply:
Let me sum this up so I won't end up lost. We have multiple ways how to solve this:

1. Define a new exception in `sysconfig` and make `distutils.error.DistutilsPlatformError` inherit from it.
2. Move `get_python_lib` and `get_python_inc` back to `distutils.sysconfig` and deprecate them together with the module.
3. Move only `get_python_lib` back to `distutils.sysconfig` and deprecate it & move `get_python_inc` to `distutils.build_ext` which is the only place where it's actually useful.
3. Keep these functions in `sysconfig` and make them return None & wrap them to raise the `DistutilsPlatformError` in `distutils.sysconfig`.

Some combinations of the points above are also possible. Do I forget something? Which one do you prefer?

Reply:
option 3 seems simple and clean to me

Reply:
Thanks @merwok. I've tried to implement point 3 but there is a problem with circular imports because of `distutils.command.build_ext` imports from `distutils.sysconfig` and `distutils.sysconfig` wants to import `get_python_inc` from `distutils.command.build_ext` to keep its backward compatibility.
I can imagine a workaround like this in `distutils.sysconfig`:
```
def get_python_inc(*args, **kwargs):
    from distutils.command.build_ext import get_python_inc
    return get_python_inc(*args, **kwargs)
```
But I'm not sure it's a good idea. What do you think?

Reply:
Then the simplest thing seems to keep these two functions in distutils.sysconfig or `__init__` or another submodule with no dependencies (on other distutils submodule).

Reply:
They are back in `distutils.sysconfig` and the exception is therefore no longer needed in `sysconfig`.